celi_framework/main.py

```diff
@@ -106,14 +136,65 @@ def bool_opt(opt: str, env_var: str, help: str):
 
     args = parser.parse_args()
 
+    # Always initialize tool_config at the beginning of the function
+    tool_config = {}  # Default to an empty dictionary or a suitable default
+
+    # # Resolve the configuration path
+    # config_path = resolve_config_path(args.tool_config_json) if args.tool_config_json else get_default_config_path() <----
+
+    # if args.tool_config_json:
+    #     config_path = resolve_config_path(args.tool_config_json)
+    # else:
+    #     # TODO - If TOOL_CONFIG_JSON=celi_framework/examples/wikipedia/example_config.json or if no .env file
+    #     # TODO- then we can do the following. but if TOOL_CONFIG_JSON= some other non-existant path then throw an error saying that you need to specific TOOL_CONFIG_JSON path in .env
+    #     config_path = get_default_config_path()
+    #
+    # # Attempt to read the configuration file
+    # if config_path and os.path.exists(config_path):
+    #     try:
+    #         config_data = read_json_from_file(config_path)
+    #         tool_config.update(config_data)  # Safely update tool_config with loaded data
+    #     except Exception as e:
+    #         print(f"Failed to read the configuration file: {e}")
+    # else:
+    #     print(f"Configuration file not found or path is invalid: {config_path}")
+    #
+    # # Check if tool_config is populated properly
+    # if not tool_config:
+    #     print("Error: Tool configuration data is missing or could not be loaded.")
+    #     return None
+
+    # Resolve and load the configuration
+    config_path = resolve_config_path(args.tool_config_json)
```

Comment:
This resolves all relative paths from the two directories above the location of the current script.  This in general will be very weird behavior for the users.  You'd expect it to resolve from the current directory.

I think it's useful to resolve to the example file int he case where none is specified, but in a general pip install for a new use case this won't ever work.